WcaOnRails/app/controllers/users_controller.rb

```diff
@@ -167,6 +171,8 @@ def update
                                  user: @user.delegate_to_handle_wca_id_claim.name)
         WcaIdClaimMailer.notify_delegate_of_wca_id_claim(@user).deliver_later
         redirect_to profile_claim_wca_id_path
+      elsif user_params.key? :subscriptions_attributes
+        redirect_to subscriptions_url
```

Comment:
Since your doing a whole new view, I think it's cleaner to have a separate controller action (eg: `update_subscriptions`, that filters only these attributes in the params, and updates only the users' subscriptions).
Technically it could be possible to craft, from the subscriptions page, a post request that updates other data from the user.